### PR TITLE
feat: allow configuring session duration

### PR DESCRIPTION
**Pull request by matthewriedel-flux**
This feature allows a user to override the default environmental session
duration (currently 1200s).

Use case: Our organisation uses CloudWatch to alert when assuming
highly-privileged roles. The default session duration generates alerts
every 20 minutes during credential rotation. This feature allows the
user to set the session duration, and for us, reduce the number of
alerts generated.

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
have the right to submit it under the open source license
indicated in the file; or

(b) The contribution is based upon previous work that, to the
best of my knowledge, is covered under an appropriate open
source license and I have the right under that license to
submit that work with modifications, whether created in whole
or in part by me, under the same open source license (unless
I am permitted to submit under a different license), as
Indicated in the file; or

(c) The contribution was provided directly to me by some other
person who certified (a), (b) or (c) and I have not modified
it.

(d) I understand and agree that this project and the contribution
are public and that a record of the contribution (including
all personal information I submit with it, including my
sign-off) is maintained indefinitely and may be redistributed
consistent with this project or the open source license(s)
involved.

Signed-off-by: Matthew Riedel <matthew.riedel@fluxfederation.com>

**Changelog**

**Bugfixes**

N/A

**Enhancements**

Allows setting a default session duration in the General options tab.

**Notes**

N/A

### CODE REVIEW

**Comment by xriu** [comment]
Would be nice to see this feature! 👍

File: src/app/models/session.ts [hunk 1]
```diff
@@ -17,12 +18,13 @@ export class Session {
...
     this.startDateTime = undefined;
+    this.duration = environment.sessionDuration;
     this.region = region;
   }
 
   expired(): boolean {
     const currentTime = new Date().getTime();
     const startTime = new Date(this.startDateTime).getTime();
-    return (currentTime - startTime) / 1000 > environment.sessionDuration;
+    return (currentTime - startTime) / 1000 > this.duration;
```
Comment (nitrocode) [inline src/app/models/session.ts:28]: If `this.duration` is unset, then shouldn't it use the `environment.sessionDuration` portion ?

Comment (matthewriedel-flux) [reply src/app/models/session.ts:28]: This PR should probably be closed- it was open prior to the 0.9.0 release which was a pretty substantial change. #283 addresses what this PR was meant to achieve, but work should probably start over in a new branch.

**Comment by matthewriedel-flux** [comment]
Closing this PR since the amount of work to rebase it with the changes in 0.9.0 would be pretty high. Seems like it would make more sense to track the work with #283 and in a new branch.

**Pull request closed without merging**